pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -0,0 +1,624 @@
+		return nil, err
+	}
+	job := &kbatch.Job{
+		ObjectMeta: meta,
+		Spec: kbatch.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: meta,
+				Spec:       pod,
+			},
+		},
+	}
+
+	if err := ctrl.SetControllerReference(vbsc, job, r.scheme); err != nil {
+		return nil, err
+	}
+
+	return job, nil
+}
+
+func (r *ReconcileVitessBackupsSchedule) createJobPod(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, name string) (pod corev1.PodSpec, err error) {
+	getVtctldServiceName := func(cluster string) (string, error) {
+		vtctldServiceName, vtctldServicePort, err := r.getVtctldServiceName(ctx, vbsc, cluster)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("--server=%s:%d", vtctldServiceName, vtctldServicePort), nil
+	}
+
+	// It is fine to not have any default in the event there is no strategy as the CRD validation
+	// ensures that there will be at least one item in this list. The YAML cannot be applied with
+	// empty list of strategies.
+	var cmd strings.Builder
+
+	addNewCmd := func(i int) {
+		if i > 0 {
+			cmd.WriteString(" && ")
+		}
+	}
+
+	for i, strategy := range vbsc.Spec.Strategy {
+		vtctldclientServerArg, err := getVtctldServiceName(vbsc.Spec.Cluster)
+		if err != nil {
+			return corev1.PodSpec{}, err
+		}
+
+		addNewCmd(i)
+		switch strategy.Name {
+		case planetscalev2.BackupShard:
+			createVtctldClientCommand(&cmd, vtctldclientServerArg, strategy.ExtraFlags, strategy.Keyspace, strategy.Shard)
+		}
+
+	}
+
+	pod = corev1.PodSpec{
+		Containers: []corev1.Container{{
+			Name:            name,
+			Image:           vbsc.Spec.Image,
+			ImagePullPolicy: vbsc.Spec.ImagePullPolicy,
+			Resources:       vbsc.Spec.Resources,
+			Args:            []string{"/bin/sh", "-c", cmd.String()},
+		}},
+		RestartPolicy: corev1.RestartPolicyOnFailure,
+		Affinity:      vbsc.Spec.Affinity,
+	}
+	return pod, nil
+}
+
+func createVtctldClientCommand(cmd *strings.Builder, serverAddr string, extraFlags map[string]string, keyspace, shard string) {
+	cmd.WriteString(fmt.Sprintf("%s %s BackupShard", vtctldclientPath, serverAddr))
+
+	// Add any flags
+	for key, value := range extraFlags {
+		cmd.WriteString(fmt.Sprintf(" --%s=%s", key, value))
+	}
+
+	// Add keyspace/shard or tablet alias
+	cmd.WriteString(fmt.Sprintf(" %s/%s", keyspace, shard))
+}
+
+func (r *ReconcileVitessBackupsSchedule) getVtctldServiceName(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, cluster string) (svcName string, svcPort int32, err error) {
+	svcList := &corev1.ServiceList{}
+	listOpts := &client.ListOptions{
+		Namespace: vbsc.Namespace,
+		LabelSelector: apilabels.Set{
+			planetscalev2.ClusterLabel:   cluster,
+			planetscalev2.ComponentLabel: planetscalev2.VtctldComponentName,
+		}.AsSelector(),
+	}
+	if err = r.client.List(ctx, svcList, listOpts); err != nil {
+		return "", 0, fmt.Errorf("unable to list vtctld service in %q: %v", vbsc.Namespace, err)
+	}
+
+	if len(svcList.Items) > 0 {
+		service := svcList.Items[0]
```

Comment:
Probably better to pick a random one to spread the load a bit:
```
service := svcList.Items[rand.IntN(len(svcList.Items))]
```

Reply:
Fixed via 54ec222